torch/csrc/autograd/variable.h

```diff
@@ -169,30 +153,31 @@ struct VariableViewImpl : public VariableImpl {
   int attr_version;
 };
 
-inline Variable make_variable(at::Tensor data, VarFlags flags=DEFAULT_FLAGS,
-                              int output_nr=0, std::shared_ptr<Function> grad_fn=nullptr) {
-  TORCH_ASSERT(!grad_fn || flags.requires_grad);
+inline Variable make_variable(at::Tensor data, bool requires_grad=false) {
   if (data.defined() && data.dim() == 0) {
     // don't expose 0-dim tensors to Variable API.
     data = data.as_strided_({1}, {1});
   }
-  return Variable(new VariableImpl(std::move(data), flags, output_nr, std::move(grad_fn)), false);
+  return Variable(new VariableImpl(std::move(data), requires_grad), false);
 }
 
-Variable make_variable(at::Tensor data, std::shared_ptr<Function> grad_fn);
-
-inline Variable make_variable(at::Tensor data, bool requires_grad, bool is_volatile=false) {
-  return make_variable(std::move(data), VarFlags(requires_grad, is_volatile));
+inline Variable make_variable(at::Tensor data, int output_nr, std::shared_ptr<Function> grad_fn) {
+  if (data.defined() && data.dim() == 0) {
+    // don't expose 0-dim tensors to Variable API.
+    data = data.as_strided_({1}, {1});
+  }
+  return Variable(new VariableImpl(std::move(data), false, output_nr, std::move(grad_fn)), false);
```

Comment:
It makes the `requires_grad()` computation a little simpler (we don't have to explicitly check if the variable is a leaf).

Reply:
Sorry, can you please elaborate on this. I'm not sure what you meant. Is it because `_requires_grad` **should be** set to `false` if the variable is not a leaf?

Reply:
I'll add an assertion, but the public API (`make_variable`) only takes `requires_grad` or only takes `grad_fn` (not both).